lib/constants.go

```diff
@@ -85,9 +85,11 @@ const (
 	NetworkType_TESTNET NetworkType = 2
 )
 
+type MsgDeSoHeaderVersion = uint32
```

Comment:
Nit: I saw this the first time and thought it's probably fine, but isn't the more normal syntax `type MsgDeSoHeaderVersion uint32` with no `=`?

Reply:
You're right. Ideally we define the type as `type MsgDeSoHeaderVersion uint32` without the `=`.

There's actually a fundamental difference between the two that makes the current definition with the `=` easier to work with for us:
- Defining the type with the `=` defines it as a type alias for `uint32`. The two are interchangeable, and values of either type can be assigned to the other. The two types are equivalent which means that all of our strongly typed uint32 byte encoding/decoding functions work as-is
- Defining the type without the `=` defines it as a brand new 4 byte unsigned integer type. Go's type system sees it as a distinct type from `uint32` type. 
  - The advantage of defining the new type is that the type system enforces usage of just the valid values of the new type. You can't assign any uint32 to the version field
  - The downside is that we would have to implement new byte encoders and decoders for every newly defined type. Or we would have to unsafe casting when encoding and decoding it

So, it seems like defining it as a type alias is the less bad approach. LMK though

Reply:
Ah wow I just read through this. Didn't know the type keyword worked like that, but makes sense. Great that we didn't have to rewrite all the code.